### PR TITLE
Tabs: await each child tab and panel update before other initialization

### LINKED ISSUE #346: Tab: First tab always gets selected disregarding the "selected" attribute
**Expected behavior**

Second tab gets selected when I add `selected` to the second tab

```html
<pharos-tab
    id="second-tab"
    data-panel-id="second-tab-panel"
    selected
>
    Second tab
</pharos-tab>
```

image

**Actual behavior**
Both the first tab and second tab are selected (the first tab always gets selected).

image

**Steps to reproduce the issue**

1. Import `initComponents` in a bundle entry point, and use the Pharos tab in a component
2. Add `selected` attribute to the second tab
3. See that both tabs are selected

**Screenshots or code**
Add this code to the root component of a bundle, you can see the issue

```jsx
<pharos-tabs>
    <pharos-tab
        id="first-tab"
        data-panel-id="first-tab-panel"
    >
        First tab
    </pharos-tab>
    <pharos-tab
        id="second-tab"
        data-panel-id="second-panel"
        selected
    >
        Second tab
    </pharos-tab>
    <pharos-tab-panel
        id="first-tab-panel"
        slot="panel"
    >
        First
    </pharos-tab-panel>
    <pharos-tab-panel
        id="second-panel"
        slot="panel"
    >
        Second
    </pharos-tab-panel>
</pharos-tabs>
```

**Pharos version**
12.3.1

**Your environment**

- OS: macOS
- Browser: Chrome
- Version: 101

**Additional information**
Some weird behavior we have found.
If the tab is directly added to the index.html, it works.

It also works if you add another random tab without a corresponding tab panel. The code below works:

```jsx
<pharos-tabs>
    <pharos-tab
        id="first-tab"
        data-panel-id="first-tab-panel"
    >
        First tab
    </pharos-tab>
    <pharos-tab
        id="first-tab"
        data-panel-id="random-panel"
    >
        Another random tab
    </pharos-tab>
    <pharos-tab
        id="second-tab"
        data-panel-id="second-panel"
        selected
    >
        Second tab
    </pharos-tab>
    <pharos-tab-panel
        id="first-tab-panel"
        slot="panel"
    >
        First
    </pharos-tab-panel>
    <pharos-tab-panel
        id="second-panel"
        slot="panel"
    >
        Second
    </pharos-tab-panel>
</pharos-tabs>
```
image

If we import the `initComponent` in the root component, it also works.

### PR DESCRIPTION
**Pull request by daneah**
**This change:** (check at least one)

- [ ] Adds a new feature
- [x] Fixes a bug
- [ ] Improves maintainability
- [ ] Improves documentation
- [ ] Is a release activity

**Is this a breaking change?** (check one)

- [ ] Yes
- [x] No

**Is the:** (complete all)

- [x] Title of this pull request clear, concise, and indicative of the issue number it addresses, if any?
- [x] Test suite(s) passing?
- [x] Code coverage maximal?
- [ ] Changeset added?
- [ ] Component status page up to date?

**What does this change address?**

Fixes #346 

**How does this change work?**

`await` the `updateComplete` for each child tab and tab panel before proceeding with component state management, ensuring they're ready by the time any action occurs on them. This is very similar to [how we manage](https://github.com/ithaka/pharos/blob/develop/packages/pharos/src/components/toggle-button-group/pharos-toggle-button-group.ts#L36) the toggle button group's tab-like functionality.

### CODE REVIEW

**Comment by daneah** [comment]
@jialin-he would you be able to `yalc` these changes and see if they fix the behavior you describe in #346?

File: packages/pharos/src/components/tabs/pharos-tabs.ts [hunk 1]
```diff
@@ -24,35 +27,37 @@ export class PharosTabs extends PharosElement {
...
+  protected override async firstUpdated(): Promise<void> {
     this.addEventListener('pharos-tab-selected', this._handleTabSelected);
     this.addEventListener('keydown', this._handleKeydown);
     this.addEventListener('focusout', this._handleFocusout);
 
-    const tabs: NodeListOf<PharosTab> = this.querySelectorAll(
-      `[data-pharos-component="PharosTab"]`
-    );
+    await Promise.all(Array.from(this._tabs).map((tab) => tab.updateComplete));
+    await Promise.all(Array.from(this._tabPanels).map((tab) => tab.updateComplete));
```
Comment (chrisjbrown) [inline packages/pharos/src/components/tabs/pharos-tabs.ts:46]: nice 👌

**Review by chrisjbrown — APPROVED** [review]

**Comment by daneah** [comment]
@jialin-he I actually gave this a try myself and it's still not quite working. Fixing it up.

**Review by jialin-he — APPROVED** [review]

**Comment by daneah** [comment]
Solving this differently; will open a new PR shortly.

**Pull request closed without merging**